src/transformers/utils/fx.py

```diff
@@ -678,114 +790,132 @@ def _generate_dummy_input(
+                inputs_dict[input_name] = torch.zeros(batch_size, long_seq_length, dtype=torch.float, device=device)
+            elif "mask" in input_name or "ids" in input_name:
+                if "mask" in input_name and "past_key_values" in input_names:
+                    inputs_dict[input_name] = torch.zeros(batch_size, past_sequence_length + sequence_length, dtype=torch.long, device=device)
                 else:
-                    raise AttributeError('Could not find the "image_size" field in the model config')
-
-            # If no num_channels is in the config, use some arbitrary value.
-            num_channels = getattr(model.config, "num_channels", 3)
-            if not isinstance(image_size, collections.abc.Iterable):
-                image_size = (image_size, image_size)
-            height, width = image_size
-            inputs_dict[input_name] = torch.zeros(
-                batch_size, num_channels, height, width, dtype=torch.float32, device=device
-            )
-        elif "bbox" in input_name:
-            inputs_dict[input_name] = torch.zeros(*shape, 4, dtype=torch.float, device=device)
-        elif "input_features" in input_name:
-            inputs_dict[input_name] = torch.zeros(
-                *shape, model.config.input_feat_per_channel, dtype=torch.float, device=device
-            )
-        elif "visual_feats" in input_name:
-            inputs_dict[input_name] = torch.zeros(
-                shape
-                + [
-                    model.config.visual_feat_dim,
-                ],
-                dtype=torch.float,
-                device=device,
-            )
-        elif "visual_pos" in input_name:
-            inputs_dict[input_name] = torch.zeros(
-                shape
-                + [
-                    model.config.visual_pos_dim,
-                ],
-                dtype=torch.float,
-                device=device,
-            )
-        elif "inputs" in input_name:
-            inputs_dict[input_name] = torch.zeros(*shape, dtype=torch.float, device=device)
-        elif "input_values" in input_name:
-            batch_size, _ = shape
-            # Generating big sequence length for audio inputs.
-            seq_length = _generate_random_int(low=10000, high=20000)
-            inputs_dict[input_name] = torch.zeros(batch_size, seq_length, dtype=torch.float, device=device)
-        elif "mask" in input_name or "ids" in input_name:
-            inputs_dict[input_name] = torch.zeros(shape, dtype=torch.long, device=device)
-        else:
-            shape_with_hidden_size = shape + [model.config.hidden_size]
-            inputs_dict[input_name] = torch.zeros(shape_with_hidden_size, dtype=torch.float, device=device)
+                    inputs_dict[input_name] = torch.zeros(shape, dtype=torch.long, device=device)
+            elif "past_key_values" == input_name:
+                if isinstance(model, BloomForCausalLM):
```

Comment:
Please use a check on the model name, like all other conditions in this file. Also, we can't start having specific code for specific models in this module or it will quickly become un-maintainable.